examples/example_piecewise_target_function.py

```diff
@@ -0,0 +1,162 @@
+"""
+Example for evolutionary regression on a piecewise target function
+==================================================================
+
+Example demonstrating the use of Cartesian genetic programming for
+regression on a piecewise target function by using the conditional (if/else) operator.
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+   Usage:
+     example_piecewise_target_function.py [--max-generations=<N>]
+
+   Options:
+     -h --help
+     --max-generations=<N>  Maximum number of generations [default: 5000]
+"""
+
+import functools
+
+import matplotlib.pyplot as plt
+import numpy as np
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+
+# %%
+# We define a piecewise target function.  The function applies different
+# transformations to the input depending whether the input is less or greater
+# than zero. Thus to achieve high fitness,
+# an individual must make use of the if/else operator.
+def f_target(x):
+    return np.select([x >= 0, x < 0], [x ** 2 + 1.0, -x])
+
+
+# %%
+# Then we define the objective function for the evolution. It uses the
+# mean-squared error between the output of the expression represented by a given
+# individual and the target function evaluated on a set of pseudo-random points.
+def objective(individual, target_function, rng):
```

Comment:
```suggestion
def objective(individual, rng):
```
i don't think we need this argument, there is only one target function anyway, isn't there?